script.embuary.helper/resources/lib/player_monitor.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+
+########################
+
+import xbmc
+import xbmcgui
+import json
+import datetime
+
+from resources.lib.helper import *
+from resources.lib.json_map import *
+from resources.lib.image import *
+
+########################
+
+class PlayerMonitor(xbmc.Monitor):
+
+    def __init__(self):
+        log('Service: Player monitor started', force=True)
+        self.fullscreen_lock = False
+        self.pvr_playback = False
+
+    def onNotification(self, sender, method, data):
+        if method in ['Player.OnPlay', 'Player.OnStop', 'Player.OnAVChange', 'Playlist.OnAdd', 'Playlist.OnRemove', 'VideoLibrary.OnUpdate', 'AudioLibrary.OnUpdate']:
+            self.data = json.loads(data)
+
+        ''' Clear music or video playlist based on player content.
+        '''
+        if method == 'Playlist.OnAdd':
+            self.clear_playlists()
+
+        ''' Get stuff when playback starts.
+        '''
+        if method == 'Player.OnPlay':
+            xbmc.stopSFX()
+            self.pvr_playback = condition('String.StartsWith(Player.Filenameandpath,pvr://)')
+
+            self.get_art_info()
+
+            if self.pvr_playback:
+                self.get_channellogo()
+
+            if PLAYER.isPlayingVideo() and not self.pvr_playback:
+                self.get_videoinfo()
+                self.get_nextitem()
+
+            if not self.fullscreen_lock:
+                self.do_fullscreen()
+
+        ''' Playlist changed. Fetch nextitem again.
+        '''
+        if method in ['Playlist.OnAdd', 'Playlist.OnRemove'] and PLAYER.isPlayingVideo() and not self.pvr_playback:
+            self.get_nextitem()
+
+        ''' Check if multiple audio tracks are available and refetch
+            artwork info for PVR playback.
+        '''
+        if method == 'Player.OnAVChange':
+            self.get_audiotracks()
+
+            if self.pvr_playback:
+                self.get_art_info()
+
+        ''' Playback stopped. Clean up.
+        '''
+        if method == 'Player.OnStop':
+            xbmc.sleep(2500)
```

Comment:
Still fine as 2500 is quite small but would be better to use the monitor

Reply:
switched to monitor